src/plugins/guided_onboarding/public/services/api.service.ts

```diff
@@ -176,55 +183,33 @@ export class ApiService implements GuidedOnboardingApi {
    * Activates a guide by guideId.
    * This is useful for the onboarding landing page, when a user selects a guide to start or continue.
    * @param {GuideId} guideId the id of the guide (one of search, kubernetes, siem)
-   * @param {GuideState} guide (optional) the selected guide state, if it exists (i.e., if a user is continuing a guide)
    * @return {Promise} a promise with the updated plugin state
    */
-  public async activateGuide(
-    guideId: GuideId,
-    guide?: GuideState
-  ): Promise<{ pluginState: PluginState } | undefined> {
-    // If we already have the guide state (i.e., user has already started the guide at some point),
-    // simply pass it through so they can continue where they left off, and update the guide to active
-    if (guide) {
-      return await this.updatePluginState(
-        {
-          status: 'in_progress',
-          guide: {
-            ...guide,
-            isActive: true,
-          },
-        },
-        true
-      );
+  public async activateGuide(guideId: GuideId): Promise<{ pluginState: PluginState } | undefined> {
```

Comment:
The logic of creating a guide state from the config if a guide is being activated for the 1st time is now moved to the server side.